dot/network/block_announce.go

```diff
@@ -191,41 +192,68 @@ func (s *Service) getBlockAnnounceHandshake() (Handshake, error) {
 	}, nil
 }
 
-func (s *Service) validateBlockAnnounceHandshake(hs Handshake) (Message, error) {
+func (s *Service) validateBlockAnnounceHandshake(peer peer.ID, hs Handshake) error {
 	var (
 		bhs *BlockAnnounceHandshake
 		ok  bool
 	)
 
 	if bhs, ok = hs.(*BlockAnnounceHandshake); !ok {
-		return nil, errors.New("invalid handshake type")
+		return errors.New("invalid handshake type")
 	}
 
 	if bhs.GenesisHash != s.blockState.GenesisHash() {
-		return nil, errors.New("genesis hash mismatch")
+		return errors.New("genesis hash mismatch")
 	}
 
 	// if peer has higher best block than us, begin syncing
 	latestHeader, err := s.blockState.BestBlockHeader()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	bestBlockNum := big.NewInt(int64(bhs.BestBlockNumber))
 
 	// check if peer block number is greater than host block number
 	if latestHeader.Number.Cmp(bestBlockNum) >= 0 {
-		return nil, nil
+		return nil
```

Comment:
store `hs Handshake` in the service's handshake data whenever this function returns nil (ie. peer's handshake is valid)
`s.ntoificationsProtocols[BlockAnnounceMsgType].handshakeData[peer].handshake = hs`

you will need to add a `handshake Handshake` field to `handshakeData` in `notifications.go`